src/Mod/PartDesign/InvoluteGearFeature.py

```diff
@@ -278,6 +278,7 @@ def reject(self):
         FreeCADGui.ActiveDocument.resetEdit()
         FreeCAD.ActiveDocument.abortTransaction()
 
-
-if FreeCAD.GuiUp:
-    FreeCADGui.addCommand('PartDesign_InvoluteGear',_CommandInvoluteGear())
+# 09/25/2024 This causes a warning when running tests, does not appear to affect access
+# to involute gears in the UI, and is a construction not found elsewhere.  Maybe left over?
+# if FreeCAD.GuiUp:
+#     FreeCADGui.addCommand('PartDesign_InvoluteGear',_CommandInvoluteGear())
```

Comment:
Without this, the command to create an involute gear will be gone -- see at the end of the "Part Design" menu from FreeCAD's menu bar.

So to get rid of the error shown in the console ("<Command> Command.cpp(1841): duplicate command PartDesign_InvoluteGear"), one could make the code not only check wether we're in GUI context, but also wether the command is already defined, e.g.
```py
if FreeCAD.GuiUp and not FreeCADGui.Command.get('PartDesign_InvoluteGear'):
    FreeCADGui.addCommand('PartDesign_InvoluteGear',_CommandInvoluteGear())
```
However, this feels like a hack.

I think the error is triggered because this very file gets imported more then once: First, [when the PartDesign WB's GUI is loaded](https://github.com/FreeCAD/FreeCAD/blob/ab36c290fd31eced07cdf8d2a81a3a489447bee6/src/Mod/PartDesign/InitGui.py#L54). And then [when the command is executed](https://github.com/FreeCAD/FreeCAD/blob/ab36c290fd31eced07cdf8d2a81a3a489447bee6/src/Mod/PartDesign/InvoluteGearFeature.py#L64) or [its tests are run](https://github.com/FreeCAD/FreeCAD/blob/ab36c290fd31eced07cdf8d2a81a3a489447bee6/src/Mod/PartDesign/PartDesignTests/TestInvoluteGear.py#L30).

Reply:
I think I found a better solution: The module which is imported to execute the feature must not add the command. Instead, the workbench should do it upon initialization: [Right where it currently imports the module](https://github.com/FreeCAD/FreeCAD/blob/ab36c290fd31eced07cdf8d2a81a3a489447bee6/src/Mod/PartDesign/InitGui.py#L54), it should setup the command -- it only imports it in order to setup the command; just indirectly. And that's the problem. It should do it directly, then we can remove the command setup from the module itself.

btw: we have the same issue with the [Sprocket feature](https://github.com/FreeCAD/FreeCAD/blob/ab36c290fd31eced07cdf8d2a81a3a489447bee6/src/Mod/PartDesign/SprocketFeature.py#L338). You just did't see it in the tests becasue this module has no tests -- which is an issue on its own.